src/matchmaking_servers.rs

```diff
@@ -0,0 +1,556 @@
+        /// * filters: An array of filters to only retrieve servers the user cares about.
+        /// A list of the keys & values can be found
+        /// [here](https://partner.steamgames.com/doc/api/ISteamMatchmakingServers#MatchMakingKeyValuePair_t).
+        ///
+        /// # Errors
+        ///
+        /// Every filter's key and value must take 255 bytes or under, otherwise `None` is returned.
+        pub fn $name<ID: Into<AppId>>(
+            &self,
+            app_id: ID,
+            filters: &HashMap<&str, &str>,
+            callbacks: ServerListCallbacks,
+        ) -> Option<Arc<Mutex<ServerListRequest>>> {
+            let app_id = app_id.into().0;
+            let mut filters = {
+                let mut vec = Vec::with_capacity(filters.len());
+                for i in filters {
+                    let key_bytes = i.0.as_bytes();
+                    let value_bytes = i.1.as_bytes();
+
+                    // Max length is 255, so 256th byte will always be nul-terminator
+                    if key_bytes.len() >= 256 || value_bytes.len() >= 256 {
+                        return None;
+                    }
+
+                    let mut key = [0i8; 256];
+                    let mut value = [0i8; 256];
+
+                    unsafe {
+                        key.as_mut_ptr()
+                            .copy_from(key_bytes.as_ptr().cast(), key_bytes.len());
+                        value
+                            .as_mut_ptr()
+                            .copy_from(value_bytes.as_ptr().cast(), value_bytes.len());
+                    }
+
+                    vec.push(sys::MatchMakingKeyValuePair_t {
+                        m_szKey: key,
+                        m_szValue: value,
+                    });
+                }
+                vec.shrink_to_fit();
+
+                vec
+            };
+
+            unsafe {
+                let callbacks = create_serverlist(callbacks);
+
+                let request_arc = ServerListRequest::get(callbacks);
+                let mut request = request_arc.lock().unwrap();
+
+                let handle = sys::$sys_method(
+                    self.mms,
+                    app_id,
+                    &mut filters.as_mut_ptr() as *mut *mut _,
+                    filters.len().try_into().unwrap(),
+                    callbacks.cast(),
+                );
+
+                request.mms = self.mms;
+                request.real = callbacks;
+                request.h_req = handle;
+
+                drop(request);
+
+                Some(request_arc)
+            }
+        }
+    };
+}
+
+pub struct GameServerItem {
+    pub appid: u32,
+    pub players: i32,
+    pub do_not_refresh: bool,
+    pub successful_response: bool,
+    pub have_password: bool,
+    pub secure: bool,
+    pub bot_players: i32,
+    pub ping: i32,
+    pub max_players: i32,
+    pub server_version: i32,
+    pub steamid: u64,
+    pub last_time_played: std::time::Duration,
```

Comment:
> time (in unix time) when this server was last played on (for favorite/history servers)
I think it would be more useful to have this as an `Instant` instead of a duration since unix epoch. Might be annoying to convert, however 🤷

Reply:
As i know there's no method to create `Instant` from milliseconds